app/controllers/bills_controller.rb

```diff
@@ -0,0 +1,52 @@
+class BillsController < ApplicationController
+  before_action :authorize_checkout, :new
+  
+  def new
+    begin
+      @session = Stripe::Checkout::Session.create({
+        payment_method_types: ['card'],
+        customer: current_user.customer_id,
+        line_items: get_line_items,
+        mode: 'payment',
+        success_url: books_url,
+        cancel_url: sales_url,
+      })
+    rescue => e
+      Rails.logger.debug e.message
+      redirect_to sales_url, notice: t('default.messages.no_item')
+    end
+    respond_to do |format|
+      format.js
+    end
+  end
+  
+  private
+    def products_params
+      params.require(:products).map {|product| product.permit(:product_id, :price)}
```

Comment:
sorry. what is the structure of the params

Reply:
`<ActionController::Parameters {"products"=>[{"price"=>"10.0", "product_id"=>"prod_HgKvRc1srJCgWA"}, {"price"=>"15.0", "product_id"=>"prod_HgKuW3VlYPZxpz"}], "controller"=>"bills", "action"=>"new"} permitted: false>`